### PR TITLE
ROX-18605: Implement Indexer Get and Has endpoints

### PR DESCRIPTION
**Pull request by jvdm**
## Description

🌳 master
├─ 🌳 ~ROX-18605: Initial conversion functions for Create Indexer Report #7043~
├─── 🌳 ~ROX-18605: Implement create index report #7077~
├───── :arrow_up_small: **ROX-18605: Implement Indexer Get and Has endpoints #7099** :point_left:

## Checklist

- [ ] Investigated and inspected CI test results
- [X] Unit test and regression tests added

### N/A

- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

## Testing Performed

UT and local `go run` calls to `GetIndexReport()` using gRPC.

### CODE REVIEW

File: scanner/services/indexer.go [hunk 1]
```diff
@@ -114,11 +128,31 @@ func validateContainerImageRequest(req *v4.CreateIndexReportRequest) error {
 	return nil
 }
 
-func (s *indexerService) GetIndexReport(ctx context.Context, req *v4.GetIndexReportRequest) (*v4.IndexReport, error) {
-	return nil, status.Error(codes.Unimplemented, "method GetIndexReport not implemented")
+// createManifestDigest creates a unique claircore.Digest from a Scanner's manifest hash ID.
+func createManifestDigest(hashID string) (claircore.Digest, error) {
```
Comment (daynewlee) [inline scanner/services/indexer.go:132]: Very minor, but maybe check if hashID is empty

**Review by daynewlee — APPROVED** [review]
Overall looks good to me

File: scanner/services/indexer.go (hunk 1 above)
Comment (RTann) [reply scanner/services/indexer.go:133]: good point, do we verify anywhere that hashID is something we'd consider valid?

File: scanner/services/indexer_test.go [hunk 2]
```diff
@@ -180,3 +180,34 @@ func (s *indexerServiceTestSuite) Test_CreateIndexReport_InvalidInput() {
 		})
 	}
 }
+
+func (s *indexerServiceTestSuite) Test_GetIndexReport() {
```
Comment (RTann) [inline scanner/services/indexer_test.go:184]: for completeness, can we add HasIndexReport tests, too?

Comment (RTann) [reply scanner/services/indexer_test.go:184]: also, nit: `TestGetIndexReport`

**Pull request merged**